cloudferrylib/os/compute/nova_compute.py

```diff
@@ -167,6 +167,12 @@ def read_info(self, target='instances', **kwargs):
                 info['instances'][instance.id] = self.convert(instance,
                                                               self.config,
                                                               self.cloud)
+                flav_details = \
+                    instances.get_flav_details(self.mysql_connector,
+                                               instance.id)
+                flav_name = self.get_flavor_from_id(instance.flavor['id']).name
+                flav_details['flav_details'].update({'name': flav_name})
+                info['instances'][instance.id]['instance'].update(flav_details)
```

Comment:
I am not sure here is a right place for this code. All such stuff we usually do in the `convert` methods. So, Is it possible to move this code block to the `convert_instance ` meth?